crypto/cmp/cmp_ctx.c

```diff
@@ -88,16 +88,38 @@ int OSSL_CMP_CTX_set1_untrusted_certs(OSSL_CMP_CTX *ctx, STACK_OF(X509) *certs)
     return 0;
 }
 
+static int cmp_ctx_set_md(OSSL_CMP_CTX *ctx, EVP_MD **pmd, int nid)
+{
+    const EVP_MD *const_md = EVP_get_digestbynid(nid);
```

Comment:
Not sure you need to do the fetch this early - you could just store the const md
and then fetch when you use it. For the CMS code - I tried to fetch just before the Init operation and then free it straight after.

Reply:
I don't remember why I switched to doing the MD object fetching in advance.
At least it has the advantage that an unsupported or wrong NID is caught early.